extra/fuegogin/context.go

```diff
@@ -0,0 +1,123 @@
+package fuegogin
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/go-fuego/fuego"
+)
+
+type ContextWithBody[B any] struct {
+	ginCtx *gin.Context
+}
+
+type ContextNoBody = ContextWithBody[any]
+
+// Body implements fuego.Ctx.
+func (c *ContextWithBody[B]) Body() (B, error) {
+	var body B
+	err := c.ginCtx.Bind(&body)
+	return body, err
+}
+
+// Context implements fuego.Ctx.
+func (c *ContextWithBody[B]) Context() context.Context {
+	return c.ginCtx
+}
+
+// Cookie implements fuego.Ctx.
+func (c *ContextWithBody[B]) Cookie(name string) (*http.Cookie, error) {
+	panic("unimplemented")
+}
```

Comment:
Define a sentinel error, you will be able to catch in tests with a type assertion, then check its value with an errors.Is


```suggestion

var ErrUnimplemented = errors.New("unimplemented")

// Cookie implements fuego.Ctx.
func (c *ContextWithBody[B]) Cookie(name string) (*http.Cookie, error) {
	panic(ErrUnimplemented)
}
```